### PR TITLE
Updated documentation of code samples

### PR DESCRIPTION
**Pull request by bergercookie**
Pull request adds summarising docstrings and explanatory messages in certain code samples provided in the /samples folder. 

I acknowledge to have: 
* Read the [`CONTRIBUTING`](https://github.com/MRPT/mrpt/blob/master/.github/CONTRIBUTING.md) page
* Updated [`docs/doxygen-pages/changelog.h`](https://github.com/MRPT/mrpt/blob/master/doc/doxygen-pages/changeLog_doc.h) to describe these changes (if applicable)
* Updated [`AUTHORS`](https://github.com/MRPT/mrpt/blob/master/AUTHORS) (if applicable)

(Notify: @MRPT/owners )

### CODE REVIEW

File: samples/icp/test.cpp [hunk 1]
```diff
@@ -7,6 +7,12 @@
    | Released under BSD License. See details in http://www.mrpt.org/License    |
    +---------------------------------------------------------------------------+ */
 
+/**
+ * Execute the iterating closest point algorithm (ICP) on a hardcoded pair of
+ * laser data. The algorithm computes the transformation (translation and
+ * rotation) for alighning the 2 sets of laser scans and plots the
```
Comment (jlblancoc) [inline samples/icp/test.cpp]: typo: aligning

File: samples/icp/test.cpp [hunk 2]
```diff
@@ -100,6 +120,13 @@ void TestICP()
 	ICP.options.dumpToConsole();
 	// -----------------------------------------------------
 
+    /**
+     * Scans alignment procedure.
+     * Given an initial guess (initialPose) and the maps to be aligned, the
+     * algorithm returns the probability density function (pdf) of the alignment
+     * Additional arguements are provided to investigate the performance of the
```
Comment (jlblancoc) [inline samples/icp/test.cpp]: typo: arguments

File: samples/poses/test.cpp [hunk 3]
```diff
@@ -68,6 +78,10 @@ void TestPoseComposition()
 	CTicTac		tictac;
 
 // ---------------------------------------------------------------
+    /**
+     * CPose3D default constructor method takes the arguements in the
```
Comment (jlblancoc) [inline samples/poses/test.cpp]: typo: arguments

**Comment by jlblancoc** [comment]
Thanks a lot for the contribution @bergercookie !! 

Please, add another commit correcting the small typos I mentioned in the code comments, and I'll merge it :+1:

**Comment by bergercookie** [comment]
Thanks! :+1:

**Pull request merged**